lib/wallet/src/Cardano/Wallet/DB/WalletState.hs

```diff
@@ -177,18 +183,59 @@ instance Buildable (DeltaWalletState1 s) where
 instance Show (DeltaWalletState1 s) where
     show = pretty
 
-{-------------------------------------------------------------------------------
-    Multiple wallets.
--------------------------------------------------------------------------------}
--- | Adjust a specific wallet if it exists or return 'ErrNoSuchWallet'.
-adjustNoSuchWallet
-    :: WalletId
-    -> (ErrNoSuchWallet -> e)
-    -> (w              -> Either e (dw, b))
-    -> (Map WalletId w -> (Maybe (DeltaMap WalletId dw), Either e b))
-adjustNoSuchWallet wid err update wallets = case Map.lookup wid wallets of
-    Nothing -> (Nothing, Left $ err $ ErrNoSuchWallet wid)
-    Just wal -> case update wal of
+adjustWallet
+    :: (w              -> Either e (dw, b))
+    -> (w -> (Maybe dw, Either e b))
+adjustWallet update wal =  case update wal of
         Left e -> (Nothing, Left e)
-        Right (dw, b) -> (Just $ Adjust wid dw, Right b)
-
+        Right (dw, b) -> (Just dw, Right b)
+
+updateStateWithResult :: (Delta da, Monad m)
+    => (Base da -> w)
+    -- ^ Get the part of the state that we want to update
+    -> DBVar m da
+    -- ^ The state variable
+    -> (w -> Either e (da, b))
+    -- ^ The update function
+    -> ExceptT e m b
+updateStateWithResult d state use =
+    ExceptT
+        $ modifyDBMaybe state
+        $ adjustWallet use . d
+
+updateState
+    :: (Delta da, Monad m)
+    => (Base da -> w)
+    -- ^ Get the part of the state that we want to update
```

Comment:
🤔 Yeah, I think that such a "selection function" is a good idea. I'm not entirely sold on the proliferation of `updatState*` — I think that function combinators which do _not_ refer to a `DBVar` are better, but I'm happy to take these function for the time being, and think them over in the future.